src/dynamic_boundary_conditions/sea_level_rise_data.py

```diff
@@ -3,106 +3,95 @@
-
-
 """
 
 from datetime import date, datetime, timedelta
 import pandas as pd
 import geopandas as gpd
 from haversine import haversine
 import os
-from shapely.geometry import Polygon, Point
+from shapely.geometry import Polygon
+from src.digitaltwin import setup_environment
 
 
-def read_slr_files(path: str):
-    """Reads in all the SLR data csv's downloaded from https://searise.takiwa.co/ into a pandas dataframe """
-    sea_level_rise_projections_nz = pd.DataFrame()
-    for filename in os.listdir(path):
+def gen_dataframe(file_path: str) -> pd.DataFrame:
+    """ read csv file from input directory."""
+    df_list = []
+    for filename in os.listdir(file_path):
         if filename.endswith('.csv'):
-            sea_level_rise_projections_region = pd.read_csv(path + filename)
-            sea_level_rise_projections_region["Region"] = filename[24:-4]
-            print('LOADED: ', filename)
-
-            sea_level_rise_projections_nz = pd.concat(
-                [sea_level_rise_projections_nz, sea_level_rise_projections_region], ignore_index=True, axis=0)
-
-    # Converting to Geopandas points
-    print('Converting to Geopandas data frame with geometry...')
-    sea_level_rise_projections_nz = gpd.GeoDataFrame(sea_level_rise_projections_nz,
-                                                     geometry=gpd.points_from_xy(sea_level_rise_projections_nz.lon,
-                                                                                 sea_level_rise_projections_nz.lat))
-
-    sea_level_rise_projections_nz.rename(
-        columns={'siteId': 'Site_ID', 'year': 'Projected_Year', 'p17': 'SLR_p17_confidence_interval_meters',
-                 'p50': 'SLR_p50_confidence_interval_meters', 'p83': 'SLR_p83_confidence_interval_meters',
-                 'lon': 'Longitude', 'lat': 'Latitude', 'measurementName': 'Scenario_(VLM_with_confidence)', },
-        inplace=True)
-
-    return sea_level_rise_projections_nz
-
-
-def unique_locations(sea_level_rise_projections_nz: gpd.geodataframe.GeoDataFrame):
-    """Finds the unique lat lon coordinates (site locations) and appends them to list and returns that list"""
-    locations_lat_lon = []
-    # Acquiring unique pairs
-    unique_location = sea_level_rise_projections_nz.geometry.unique()
-    # Adding pairs to list of dictionarys to map to later on
-    print('Finding unique locations...')
-    for i in unique_location:
-        locations_value_pair = {'lat': i.y, 'lon': i.x}
-        locations_lat_lon.append(locations_value_pair)
-    return locations_lat_lon
+            try:
+                print('LOADED CSV file by pyarrow: ', filename)
+                #  pyarrow is faster, need Pandas 1.4, released in January 2022
+                df = pd.read_csv(file_path + filename, engine='pyarrow', dtype={'siteId': int})
+            except:
+                print('LOADED CSV file: ', filename)
+                df = pd.read_csv(file_path + filename, dtype={'siteId': int})
```

Comment:
I like this method of retrying. I would change it so the print call comes after the reading though
```suggestion
            try:
                #  pyarrow is faster, need Pandas 1.4, released in January 2022
                df = pd.read_csv(file_path + filename, engine='pyarrow', dtype={'siteId': int})
                print(f"LOADED {filename} by pyarrow")
            except:
                df = pd.read_csv(file_path + filename, dtype={'siteId': int})
                print(f"LOADED {filename}")
```

Reply:
Yes, it should be under the potential wrong line. Already committed.